src/utils/chains.ts

```diff
@@ -0,0 +1,362 @@
+import { Address, BigDecimal, BigInt } from '@graphprotocol/graph-ts'
+
+import { OPTIMISM_POOL_MAPPINGS } from '../backfill/poolMappings'
+import { StaticTokenDefinition } from './staticTokenDefinition'
+
+export enum ChainId {
+  ARBITRUM_ONE = 42161,
+  AVALANCHE = 43114,
+  BASE = 8453,
+  BLAST_MAINNET = 81457,
+  BSC = 56,
+  CELO = 42220,
+  MAINNET = 1,
+  MATIC = 137,
+  OPTIMISM = 10,
+}
+
+// Note: All token and pool addresses should be lowercased!
+export class SubgraphConfig {
+  // deployment address
+  // e.g. https://docs.uniswap.org/contracts/v3/reference/deployments/ethereum-deployments
+  factoryAddress: string
+
+  // the address of a pool where one token is a stablecoin and the other is a
+  // token that tracks the price of the native token use this to calculate the
+  // price of the native token, so prefer a pool with highest liquidity
+  stablecoinWrappedNativePoolAddress: string
+
+  // true is stablecoin is token0, false if stablecoin is token1
+  stablecoinIsToken0: boolean
+
+  // the address of a token that tracks the price of the native token, most of
+  // the time, this is a wrapped asset but could also be the native token itself
+  // for some chains
+  wrappedNativeAddress: string
+
+  // the mimimum liquidity in a pool needed for it to be used to help calculate
+  // token prices. for new chains, this should be initialized to ~4000 USD
+  minimumNativeLocked: BigDecimal
+
+  // list of stablecoin addresses
+  stablecoinAddresses: string[]
+
+  // a token must be in a pool with one of these tokens in order to derive a
+  // price (in addition to passing the minimumEthLocked check). This is also
+  // used to determine whether volume is tracked or not.
+  whitelistTokens: string[]
+
+  // token overrides are used to override RPC calls for the symbol, name, and
+  // decimals for tokens. for new chains this is typically empty.
+  tokenOverrides: StaticTokenDefinition[]
+
+  // skip the creation of these pools in handlePoolCreated. for new chains this is typically empty.
+  poolsToSkip: string[]
+
+  // initialize this list of pools and token addresses on factory creation. for new chains this is typically empty.
+  poolMappings: Array<Address[]>
+}
+
+export function getSubgraphConfig(): SubgraphConfig {
+  // Update this value to the corresponding chain you want to deploy
+  const SELECTED_CHAIN = ChainId.BASE
```

Comment:
👍🏼 to putting it in the readme 
> 1. Create a new subgraph config in `src/utils/chains.ts`. This will require adding a new `ChainId`. Set the `SELECTED_CHAIN` in `getSubgraphConfig()` to be this new `ChainId`.

found this, should be able to refactor https://ethereum.stackexchange.com/questions/137807/how-to-develop-a-network-or-chain-agnostic-subgraph